README.md

```diff
@@ -1,6 +1,239 @@
-# ravendb-ruby-client
+# RavenDB client for Ruby
+
+## Installation
+
+For global intsall run:
+
+```bash
+gem install --save ravendb
+```
+
+Or require `ravendb` gem in your project's `Gemfile`:
+
+```
+source 'https://rubygems.org'
+gem 'ravendb'
+```
+
+## Getting started
+
+1. Declare document models as classes. Attributes should be just an instance variables accessible via `attr_accessor()`
+
+```ruby
+class Product
+  :attr_accessor :id, :title, :price, :currency,
+                 :storage, :manufacturer, :in_stock,
+                 :last_update
+  
+  def initialize(
+    id = nil,
+    title = '',
+    price = 0,
+    currency = 'USD',
+    storage = 0,
+    manufacturer = '',
+    in_stock = false,
+    last_update = nil
+  )
+    @id = id
+    @title = title
+    @price = price
+    @currency = currency
+    @storage
+    @manufacturer = manufacturer
+    @in_stock = in_stock
+    @last_update = last_update
+  end
+end
+```
+
+2. Require `ravendb` gem and models classes in your application:
+
+```ruby
+require 'ravendb'
+require 'models/product'
+```
+
+3. Configure document store:
+
+```ruby
+RavenDB.store.configure do |config|
+  config.urls = ["database url"]
+  config.default_database = 'default database name'
+end
+```
+
+3. Open a session:
+
+```ruby
+session = RavenDB.store.open_session
+```
+
+4. Do some actions with documents:
+
+```ruby
+product = session.load('Products/1-A')
+product.in_stock = true
+
+session.store(product)
```

Comment:
Why is this required if we loaded it from the server?

Reply:
@ayende In NodeJS / Ruby / Python client we need to call store() for notify session that doc was changed

Reply:
Why? Aren't we doing change tracking there?

Reply:
At least in Python client, we _are_ doing that, and this is what we want to have

Reply:
@ayende we're doing change tracking but store also performs some additional checks + updates cached document info so anyway it should be called.
see how its done in Python client - i use it as my codebase

Reply:
so why is the `store` needed?

Reply:
- it builds default metadata for new documents (wasn't loaded before from this session)
- sets expected change vector(if was passed to store()) according to concurrency check mode
- checks if we already called store for this document
- checks if we already called delete for this document. documents marked as deleted can't be stored
- generates id (if document was new)
- updates current document state (raw entity + metadata + change vector)  in cache

Reply:
* Why do we need to build metadata? We already loaded the document, and it had the metadata there.
* Setting the change vector is understood and how it works elsewhere.
* I'm talking specifically about the scenario in the doc here, where you are calling `store` after `load.

Reply:
In the scenario in the doc here, where we are calling store after load, the store() method do the following thinks:
- sets expected change vector (if was passed to store()) according to concurrency check mode
- checks if we already called store for this document
- checks if we already called delete for this document. documents marked as deleted can't be stored
- updates current document state (raw entity + metadata + change vector) in cache

Reply:
Is this required? What happens if we'll drop it, will everything still work?

Reply:
@ayende I think better to ask this question to Idan as he's an author of Python client. I just took this logic and ported it to nodeJS. After i did it, there were just small changes (related to change vectors handling)

Reply:
@ayende From my point of view it's just a part of Raven concept which prefers short sessions with less amount of requests. So send delete & update or double/triple updates of the same document are forcibly denied

Reply:
I spoke to @IdanHaim about this, according to him, this is not required.

Reply:
And yes, short sessions are preferred, but the process is usually:
`load`, modify, `save_changes`